server/risk_scores/risk_assessment.py

```diff
@@ -65,43 +69,117 @@ def get_incident_similarity(prev_incident: submit_schema.Form, current_incident:
+
+    max_total_prev_risk_score = max_prev_risk_score
+    return total_prev_risk_score / max_total_prev_risk_score
+
+
+def normalize_current_risk_score(risk_score: float):
+    """
+    Normalizes the risk_score by the maximum potential incident risk score, returning a value between 0 and 1.
+    Params:
+        risk_score: Risk score based on the Critical Incident Report.
+    """
+    return risk_score / risk_scores.max_risk_score
+
+
+def calculate_previous_risk_score_weightings() -> List[float]:
+    """
+    Creates a risk score weighting distribution of size MAX_PREVIOUS_INCIDENTS such that the distribution
+    is a decreasing linear series that sums to 1. For example, with MAX_PREVIOUS_INCIDENTS == 3,
+    this function returns [0.5, 0.333..., 0.166...], or [3/6, 2/6. 1/6].
+
+    This is designed to put more weighting on the first most recent incident types, as well
+    as lessen the difference between the number of incident types a client has.
+    """
+    mpi = MAX_PREVIOUS_INCIDENTS
+    # arithmetic sum formula
+    denominator = mpi * (mpi+1) / 2
+
+    return [(mpi-i)/denominator for i in range(mpi)]
+
+
+def get_previous_incidents_risk_score(form: submit_schema.Form, timeframe: int):
+    """
+    Returns a risk score number between 0 and 1 based on the last MAX_PREVIOUS_INCIDENTS by that client
+    with the same primary initials in the database.
+
+    Params:
+        form: Data submitted in through the endpoint.
+        timeframe: Number of months to search over, i.e., months before previous incidents become irrelevant.
+    """
     query = {
         "client_primary": form.client_primary,
         "occurrence_time": {
-            '$gte': (form.occurrence_time - relativedelta(months=timeframe)).strftime("%Y-%m-%d %H:%M:%S")
+            "$gte": (form.occurrence_time - relativedelta(months=timeframe)).strftime("%Y-%m-%d %H:%M:%S")
         }
     }
-    prev_incidents = collection.find(query)
+    sort_order = [("occurrence_time", -1)]
+    prev_incidents = list(collection.find(query).sort(
+        sort_order))[:MAX_PREVIOUS_INCIDENTS]
+
+    previous_risk_score_weightings = calculate_previous_risk_score_weightings()
+
     total_prev_risk_score = 0
-    for incident_dict in prev_incidents:
+    for i, incident_dict in enumerate(prev_incidents):
         incident_dict = {key: (val.lower() if type(val) == str else val)
                          for key, val in incident_dict.items()}
-
         incident = Form(**incident_dict)
-        incident_score = get_current_risk_score(incident)
-        incident_similarity = get_incident_similarity(incident, form)
-        incident_recency = get_incident_recency(incident, form, timeframe)
-
-        total_prev_risk_score += incident_score + \
-            incident_recency * incident_similarity
+        total_prev_risk_score += normalize_previous_risk_score(get_previous_incident_risk_score(
+            form, incident, 1)) * previous_risk_score_weightings[i]
```

Comment:
Why do we need to use different weights for older incidents? Lower weights for older incidents is already accounted for in the `incident_recency` multiplier in `get_previous_incident_risk_score`. This seems like it will lower older incidents even more, just based on the order of their `occurrence_time`. Maybe we can just take a linear average instead, and adjust the weights in `get_incident_recency` if needed?

Reply:
The timeframe should just be passed in here probably, good catch too

Reply:
My reasoning was more so that I wanted to minimize the risk score difference between the number of previous incident types for each new report. e.g., the weightings of 2 previous risk scores would be `3/6 + 2/6 = 5/6` instead of `1/3+1/3 = 4/6`. I do agree that the recency plays a factor in all of this, but less so if the latest previous incidents are closer together.